@i-be-snek , I would like to request support for the normalization step. I encountered issues with cases such as “dozen” and “many” — the original code does not handle these correctly. I tested this in a new branch and observed the same failures as in my current branch. They passed the test in **test__extract_approximate_quantifiers**, but not working for the **extract_numbers**. Could we work together on this on Friday? thanks! 
```
FAILED tests/test_normalize_numbers.py::TestNormalizeNumbers::test__extract_simple_range[twelve and one hundred-None] - AssertionError: assert (12, 100) == None
FAILED tests/test_normalize_numbers.py::TestNormalizeNumbers::test_extract_numbers[several dozen people-expected28] - assert (0, 0, 0) == (24, 72, 1)
FAILED tests/test_normalize_numbers.py::TestNormalizeNumbers::test_extract_numbers[more than eighty-expected30] - assert (0, 0, 1) == (81, 91, 1)
FAILED tests/test_normalize_numbers.py::TestNormalizeNumbers::test_extract_numbers[more than two million-expected33] - assert (1000000, 1000000, 1) == (2000001, 3000001, 1)
FAILED tests/test_normalize_numbers.py::TestNormalizeNumbers::test_extract_numbers[a dozen deaths were reported-expected34] - assert (0, 0, 0) == (12, 12, 1)
FAILED tests/test_normalize_numbers.py::TestNormalizeNumbers::test_extract_numbers[dozens of hundreds of homes were completely destroyed-expected35] - assert (0, 0, 0) == (2400, 7200, 1)
FAILED tests/test_normalize_numbers.py::TestNormalizeNumbers::test_extract_numbers[many were killed!-expected36] - assert (0, 0, 0) == (20, 60, 1)
FAILED tests/test_normalize_numbers.py::TestNormalizeNumbers::test_extract_numbers[Only a number of victims were found-expected37] - assert (0, 0, 0) == (2, 6, 1)

--- a/.github/workflows/run_tests.yml
+++ b/.github/workflows/run_tests.yml
@@ -7,6 +7,7 @@ on:
   pull_request:
     branches:
       - main
+      - WikimpactsV2
 
 jobs:
   test:

--- a/Database/scr/normalize_numbers.py
+++ b/Database/scr/normalize_numbers.py
@@ -233,7 +233,7 @@ class NormalizeNumber:
                 "tril": "trillion",
             },
         }
-        text = regex.sub(r"\[[^\]]*]", " ", text) # drop the citation brackets such as [1], [23], (note) …
+        text = regex.sub(r"\[[^\]]*]", " ", text)  # drop the citation brackets such as [1], [23], (note) …
         # remove currency
         text = " ".join(regex.sub(r"\p{Sc}|(~)|Rs\.|Rs", " \\g<1> ", text).split())
 
@@ -478,26 +478,25 @@ class NormalizeNumber:
                 return 0
 
             return 0
-    
+
     def _extract_simple_range(self, text: str) -> Tuple[float, float] | None:
         sep = "-"
-        for i in ("and", "to", "&","or"): # add "or" in the sep sign
+        for i in ("and", "to", "&", "or"):  # add "or" in the sep sign
             if i in text:
                 sep = i
                 break
         try:
             nums = [x.replace(",", "") for x in text.split(sep)]
             if len(nums) == 2:
-                try: # if the num is not digits, try to convert the word to digit 
-                    
+                try:  # if the num is not digits, try to convert the word to digit
                     return (self.atof(nums[0].strip()), self.atof(nums[1].strip()))
-                except :
+                except:
                     try:
                         # Fallback: try to extract single numbers manually
                         left_num = self._extract_single_number(nums[0])
                         right_num = self._extract_single_number(nums[1])
                         return (left_num[0], right_num[0])
-                    except :
+                    except:
                         return None
         except:
             # try again but first normalize the number first
@@ -506,16 +505,16 @@ class NormalizeNumber:
             if len(nums) == 2:
                 try:
                     return (self.atof(nums[0].strip()), self.atof(nums[1].strip()))
-                except :
+                except:
                     try:
                         # Fallback: try to extract single numbers manually
                         left_num = self._extract_single_number(nums[0])
                         right_num = self._extract_single_number(nums[1])
                         return (left_num[0], right_num[0])
-                    except :
+                    except:
                         return None
         return None
-    
+
     def _get_scale(self, n_init: float | int):
         """
         Determine the scale of a number
@@ -742,17 +741,13 @@ class NormalizeNumber:
             approx = 1
         except BaseException:
             try:
-          
                 numbers = self._extract_complex_range(text)
-               
                 assert numbers, BaseException
                 approx = 1
             except BaseException:
                 try:
                     cleaned_text = " ".join(regex.sub(r"\s+[A-Z]{1,3}\s+", " ", text).split())
-                
                     numbers = self._extract_complex_range(cleaned_text)
-                
                     assert numbers, BaseException
                     approx = 1
                 except BaseException:

--- a/tests/test_normalize_numbers.py
+++ b/tests/test_normalize_numbers.py
@@ -136,10 +136,12 @@ class TestNormalizeNumbers:
             ("0 - 352", (0, 352)),
             ("23- 55", (23, 55)),
             ("24,501-61,672", (24501, 61672)),
+            ("twelve and one hundred", (12, 100)),
+            ("seven to thirteen", (7, 13)),
+            ("12 & 700", (12, 700)),
             # cases meant to fail
             (">=12", None),
             ("12", None),
-            ("twelve and one hundred", None),
         ],
     )
     def test__extract_simple_range(self, test_input, expected):
@@ -186,12 +188,11 @@ class TestNormalizeNumbers:
             ("more than eighty", (81, 91, 1)),
             ("ten to eleven", (10, 11, 1)),
             ("ten[2] or eleven", (10, 11, 1)),
-            ("more than two million", (2000001,3000001 , 1)),
-            ("a dozen deaths were reported", (12, 12,1)),
-            ("dozens of hundreds of homes were completely destroyed", (2400, 7200,1)),
-            ("many were killed!", (20, 60,1)),
-            ("Only a number of victims were found", (2, 6,1)),
-          
+            ("more than two million", (2000001, 3000001, 1)),
+            ("a dozen deaths were reported", (12, 12, 1)),
+            ("dozens of hundreds of homes were completely destroyed", (2400, 7200, 1)),
+            ("many were killed!", (20, 60, 1)),
+            ("Only a number of victims were found", (2, 6, 1)),
         ],
     )
     def test_extract_numbers(self, test_input, expected):

